### PR TITLE
[feat] Add pytorch lightning epoch tracking

### PR DESCRIPTION
**Pull request by tmynn**
Added epoch value to `track` method.

### CODE REVIEW

File: aim/sdk/adapters/pytorch_lightning.py [hunk 1]
```diff
@@ -111,7 +111,12 @@ def log_metrics(self, metrics: Dict[str, float],
         assert rank_zero_only.rank == 0, \
             'experiment tried to log from global_rank != 0'
 
-        for k, v in metrics.items():
+        metric_items: Dict[str: Any] = {k:v for k, v in metrics.items()}
+
+        if 'epoch' in metric_items:
+            epoch: int  = metric_items.pop('epoch')
```
Comment (alberttorosyan) [inline aim/sdk/adapters/pytorch_lightning.py:117]: should we add?
```python
else:
    epoch = None
```
to cover all the cases?

**Review by alberttorosyan — APPROVED** [review]

**Comment by tmynn** [comment]
Resolves #1957

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,11 @@
 
 ## Unreleased
 
+### Enhancements:
+
 - Implement vertically scalable version of Remote Tracking (mihran113, alberttorosyan)
+- Add Optuna integration (tmynn)
```
Comment (mihran113) [inline CHANGELOG.md:8]: this was released with `3.14.0` right?

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +30,7 @@
 - Create a Dashboard page to provide a better onboarding experience (arsengit, roubkar, KaroMourad, mihran113)
 - Add support for tracking jax device arrays (mihran113)
 - Add Optuna integration (tmynn)
+- Add epoch tracking for PyTorch Lightning (tmynn)
```
Comment (mihran113) [inline CHANGELOG.md:33]: this shouldn't be here

**Review by mihran113 — APPROVED** [review]
some issues in changelog, the rest looks good

**Pull request merged**